chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,606 @@
+from __future__ import annotations
+
+import json
+import logging
+import time
+from typing import Any, Dict, List, Optional, Set, Tuple
+
+from chia_rs import G1Element, G2Element
+from typing_extensions import Unpack
+
+from chia.protocols.wallet_protocol import CoinState
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.serialized_program import SerializedProgram
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_spend import CoinSpend, make_spend
+from chia.types.signing_mode import SigningMode
+from chia.types.spend_bundle import SpendBundle
+from chia.util.ints import uint32, uint64, uint128
+from chia.wallet.coin_selection import select_coins
+from chia.wallet.conditions import Condition, CreateCoin, CreatePuzzleAnnouncement, parse_timelock_info
+from chia.wallet.derivation_record import DerivationRecord
+from chia.wallet.lineage_proof import LineageProof
+from chia.wallet.payment import Payment
+from chia.wallet.puzzles.p2_conditions import puzzle_for_conditions, solution_for_conditions
+from chia.wallet.signer_protocol import (
+    PathHint,
+    SignedTransaction,
+    SigningInstructions,
+    SigningResponse,
+    Spend,
+    SumHint,
+)
+from chia.wallet.transaction_record import TransactionRecord
+from chia.wallet.util.compute_hints import compute_spend_hints_and_additions
+from chia.wallet.util.transaction_type import TransactionType
+from chia.wallet.util.tx_config import CoinSelectionConfig, TXConfig
+from chia.wallet.util.wallet_sync_utils import fetch_coin_spend
+from chia.wallet.vault.vault_drivers import (
+    construct_p2_delegated_secp,
+    construct_secp_message,
+    construct_vault_merkle_tree,
+    get_p2_singleton_puzzle,
+    get_p2_singleton_puzzle_hash,
+    get_recovery_finish_puzzle,
+    get_recovery_inner_puzzle,
+    get_recovery_puzzle,
+    get_recovery_solution,
+    get_vault_full_puzzle,
+    get_vault_full_solution,
+    get_vault_hidden_puzzle_with_index,
+    get_vault_inner_puzzle,
+    get_vault_inner_puzzle_hash,
+    get_vault_inner_solution,
+    get_vault_proof,
+)
+from chia.wallet.vault.vault_info import RecoveryInfo, VaultInfo
+from chia.wallet.vault.vault_root import VaultRoot
+from chia.wallet.wallet import Wallet
+from chia.wallet.wallet_info import WalletInfo
+from chia.wallet.wallet_protocol import GSTOptionalArgs
+
+
+class Vault(Wallet):
+    @staticmethod
+    async def create(
+        wallet_state_manager: Any,
+        info: WalletInfo,
+        name: str = __name__,
+    ) -> Vault:
```

Comment:
I see `.vault_info` in other parts of the program, but I don't see it being set here.  Could we potentially make this wallet a dataclass so that all of the properties unique to it are at the top and are always set?